src/main/java/com/dutact/web/features/event/admin/controllers/AdminEventController.java

```diff
@@ -1,42 +1,113 @@
 package com.dutact.web.features.event.admin.controllers;
 
+import com.dutact.web.auth.context.SecurityContextUtils;
+import com.dutact.web.auth.factors.OrganizerService;
+import com.dutact.web.auth.factors.Role;
 import com.dutact.web.core.entities.event.EventStatus;
+import com.dutact.web.features.event.admin.dtos.EventCreateDto;
 import com.dutact.web.features.event.admin.dtos.EventDto;
+import com.dutact.web.features.event.admin.dtos.EventUpdateDto;
 import com.dutact.web.features.event.admin.services.EventService;
+import lombok.SneakyThrows;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
+import java.net.URI;
 import java.util.Collection;
-import java.util.Optional;
 
 @RestController
 @RequestMapping("/api/admin/events")
 public class AdminEventController {
 
-    public EventService eventService;
+    private final EventService eventService;
+    private final OrganizerService organizerService;
 
-    public AdminEventController(EventService eventService) {
+    public AdminEventController(EventService eventService, OrganizerService organizerService) {
         this.eventService = eventService;
+        this.organizerService = organizerService;
+    }
+
+    @PostMapping
+    @SneakyThrows
+    @ResponseStatus(HttpStatus.CREATED)
+    public ResponseEntity<EventDto> createEvent(@RequestBody EventCreateDto eventDto) {
+        if (!SecurityContextUtils.hasRole(Role.EVENT_ORGANIZER)) {
```

Comment:
Do we really need to validate the organizer to create event, I see no any case for this to happen

Reply:
So why not?

Reply:
Can Admin and student affair office create event?